flamedisx/source.py

```diff
@@ -215,8 +215,6 @@ def set_defaults(self, **params):
             if k in self.defaults:
                 self.defaults[k] = tf.convert_to_tensor(
                     v, dtype=fd.float_type())
-            else:
-                raise ValueError(f"Key {k} not in defaults")
```

Comment:
Is this still needed now that we pass params filtered by `defaults_in_sources[sname]`?

Reply:
It is needed for the simulation as it takes all the fit params without distinction; the if checks if it's in the defaults and if not doesn't do anything. with the error there, if the parameter is not in the defaults for the specific source it will raise an error even if the parameter is not given then to the source.

Reply:
Ok, thanks